.github/workflows/wheel_tests_and_release.yml

```diff
@@ -48,6 +45,26 @@ jobs:
           python -m build --no-isolation --skip-dependency-check --sdist .
           ls -la ${{ github.workspace }}/dist
 
+      - name: Generate artifact attestation for sdist and wheels
+        uses: actions/attest-build-provenance@173725a1209d09b31f9d30a3890cf2757ebbff0d # v1.1.2
+        with:
+          subject-path: "dist/scikit_image-*"
+
+      - name: Verify the distribution
+        run: pipx run twine check --strict dist/*
+
+      # Ensure that a compromised twine couldn't have altered the distributions
+      # Required to resolved sdist and whl separately
+      - name: Verify sdist artifact attestation
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: gh attestation verify dist/scikit_image-*.tar.gz --repo ${{ github.repository }}
+
+      - name: Verify wheel artifact attestation
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: gh attestation verify dist/scikit_image-*.whl --repo ${{ github.repository }}
```

Comment:
Instead of manually doing checksum validation, validate with `twine` _after_ the attestations are created, and then use the attestations just created and published to validate that the distributions are unaltered before upload to PyPI.

Reply:
YES, that is so cool :sunglasses: